optimum/neuron/pipelines/transformers/base.py

```diff
@@ -0,0 +1,238 @@
+        model_id = model
+        neuronx_model_class = SUPPORTED_TASKS[targeted_task]["class"][0]
+        model = neuronx_model_class.from_pretrained(model, export=export, **model_kwargs, **input_shapes)
+    # uses neuron model
+    elif isinstance(model, NeuronBaseModel):
+        if tokenizer is None and load_tokenizer:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, (PreTrainedTokenizer, PreTrainedTokenizerFast)):
+                    tokenizer = preprocessor
+                    break
+            if tokenizer is None:
+                raise ValueError(
+                    "Could not automatically find a tokenizer for the NeuronBaseModel, you must pass a tokenizer explictly"
+                )
+        if feature_extractor is None and load_feature_extractor:
+            for preprocessor in model.preprocessors:
+                if isinstance(preprocessor, SequenceFeatureExtractor):
+                    feature_extractor = preprocessor
+                    break
+            if feature_extractor is None:
+                raise ValueError(
+                    "Could not automatically find a feature extractor for the ORTModel, you must pass a "
+                    "feature_extractor explictly"
+                )
+        model_id = None
+    else:
+        raise ValueError(
+            f"""Model {model} is not supported. Please provide a valid model either as string or NeuroModel.
+            You can also provide non model then a default one will be used"""
+        )
+    return model, model_id, tokenizer, feature_extractor
+
+
+def pipeline(
+    task: str = None,
+    model: Optional[Any] = None,
+    tokenizer: Optional[Union[str, PreTrainedTokenizer]] = None,
+    feature_extractor: Optional[Union[str, PreTrainedFeatureExtractor]] = None,
+    use_fast: bool = True,
+    export: bool = False,
+    input_shapes: Optional[Dict[str, int]] = None,
+    use_auth_token: Optional[Union[str, bool]] = None,
+    revision: Optional[str] = None,
+    trust_remote_code: Optional[bool] = None,
+    *model_kwargs,
+    **kwargs,
+) -> Pipeline:
+    if task not in list(NEURONX_SUPPORTED_TASKS.keys()):
+        raise ValueError(
+            f"Task {task} is not supported for the optimum neuron pipeline. Supported tasks are {list(NEURONX_SUPPORTED_TASKS.keys())}"
+        )
+
+    if input_shapes is None and export:
+        input_shapes = {"batch_size": 1, "sequence_length": 128}
+        logger.warning(f"No input shapes provided, using default shapes, {input_shapes}")
+
+    # copied from transformers.pipelines.__init__.py
+    hub_kwargs = {
+        "revision": revision,
+        "use_auth_token": use_auth_token,
+        "trust_remote_code": trust_remote_code,
+        "_commit_hash": None,
+    }
+
+    config = kwargs.get("config", None)
+    if config is None and isinstance(model, str):
+        config = AutoConfig.from_pretrained(model, _from_pipeline=task, **hub_kwargs, **kwargs)
```

Comment:
So "**kwargs" are keyword arguments to pass to the config constructor only?

Reply:
copied from optimum